scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+    if [ ! "$NETCAT_KILL" == "" ];then
+        wsrep_log_info "-> NetCat still up PID: $NETCAT_KILL. Will try to kill it"
+        kill -9 $NETCAT_KILL
+        if [ "$?" != "0" ]; then
+            wsrep_log_debug "-> Cannot kill the existing NetCat process PID $NETCAT_KILL"
+        fi
+    fi
+
+    NC_PID=""
+
+    # No data dir, need to initialize one first, to make connections to
+    # this node possible.
+    # We need to use a temporary empty data directory, because the
+    # actual datadir may already contain some wsrep-related files
+    wsrep_log_info "Initializing data directory at $tmp_datadir"
+
+    wsrep_log_info "INITIALIZE DB"
+
+    echo "" > $CLONE_ERR
+    eval $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS \
+         $SKIP_NETWORKING_OPTIONS \
+         --initialize-insecure --wsrep_provider=none --datadir="$tmp_datadir" >> $CLONE_ERR 2>&1 || \
+    ( wsrep_log_error "Failed to initialize data directory. Some hints below:"
+      grep '[ERROR]' $CLONE_ERR | cut -d ']' -f 4- | while read msg
+      do
+          wsrep_log_error "> $msg"
+      done
+      wsrep_log_error "Full log at $CLONE_ERR"
+      exit 1 )
+
+    # Move initialized data directory structure to real datadir and cleanup
+    mv --force "$tmp_datadir"/* "$WSREP_SST_OPT_DATA/"
+    sleep 2
+    wsrep_log_debug "-> REMOVE $tmp_datadir"
+    rm -rf "$tmp_datadir"
+
+    wsrep_log_info "INITIALIZE DB DONE"
+
+    # Need to create an extra user for 'localhost' because in some installations
+    # by default exists user ''@'localhost' and shadows every user with '%'.
+ cat << EOF > "$CLONE_SQL"
+ SET SESSION sql_log_bin=OFF;
+ CREATE USER '$CLONE_USER'@'%' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* to '$CLONE_USER'@'%';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER';
+ CREATE USER '$CLONE_USER'@'localhost' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* TO '$CLONE_USER'@'localhost';
+ GRANT EXECUTE ON *.* to '$CLONE_USER'@'localhost';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER'@'localhost';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER'@'localhost';
+EOF
+
+    wsrep_log_info "Launching clone recipient daemon"
+    wsrep_log_info "-> using: $CLONE_ENV $CLONE_BINARY_SAFE $DEFAULT_OPTIONS "
+    wsrep_log_debug "-> Test connection as: -u$CLONE_USER -pxxxxxx -h $JOINER_CLONE_HOST -P $JOINER_CLONE_PORT"
+
+    # Define client to be used on the Joiner side
+    MYSQL_ACLIENT="$MYSQL_CLIENT -u$CLONE_USER -S$CLONE_SOCK --batch --skip_column_names --silent"
+    wsrep_log_debug "-> MYSQL_ACLIENT: $MYSQL_ACLIENT"
+
+    # HERE We start the instance that will get the clone and pass the CLONE_SQL file to create user
+    eval WSREP_SST_CLONE=1 $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS \
```

Comment:
Where is the env variable `WSREP_SST_CLONE=1` used? If it is needed, please add a comment saying why it is needed

Reply:
Removed.